### PR TITLE
Relaycode - An Improved Extrinsics Builder for Polkadot

### PR DESCRIPTION
**Pull request by itsyogesh**
#### Relaycode - An Improved Extrinsics Builder for Polkadot

>Relaycode is an advanced extrinsics building platform that enhances the user experience for developers and users in the Polkadot ecosystem. It provides an intuitive interface for building and analyzing extrinsics, making it easier for both novice and experienced users to interact with the network.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (DOT, USDC & USDT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by itsyogesh** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @itsyogesh sorry if I forgot to mention this during our call, but we ask all grantees to take a percentage of the total costs in vested DOT. The DOT is linearly vested on-chain for two years. Would you be willing to accept some of the payment in DOT? If so, please add a `DOT %` line to indicate the amount of vested DOT you are willing to receive. See [this example](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md?plain=1#L127). The recommendation is 50%, but you can argue for less.

**Comment by itsyogesh** [comment]
Hi @keeganquigley thank you for clarifying the requirements. I've updated the application with the overall vesting percentage as well as percentage per milestone.

>Note - The budget was adjusted to account for unforeseen costs including vested DOT

**Comment by PieWol** [comment]
Whats the difference between your proposed solution and the possibility to craft extrinsics on polkadot.js/apps?

**Comment by itsyogesh** [comment]
Hi @PieWol,  so there are a couple of differences when we compare Relaycode to polkadot.js/apps:

1. The creation of extrinsics and decoding happens in the same view compared to different tabs in the the extrinsics page of polkadot.js/apps.
2. Bi-directional editing of values: You can seamlessly switch between editing readable values and raw encoded data. Changes are instantly reflected in both sections. This is especially needed when you have to submit multiple extrinsics with just a slight argument change.
3. Built-in guides for each pallet and method to help users understand them better. Users will also be able connect their wallet directly to sign and submit extrinsics.

We are also writing this with full typescript support including the scale codec with [Dedot](https://github.com/dedotdev/dedot) along with SSR implementation.

**Comment by keeganquigley** [comment]
Thanks @itsyogesh looks good as far as formatting goes; I will mark the application as ready for review so the rest of the committee can comment.

File: applications/relaycode.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+3. Wallet integration for seamless signing and submission
+4. Comprehensive documentation for each pallet and guides
+
+Relaycode integrates directly with Substrate-based chains, focusing initially on Polkadot, moving on to Kusama and other parachains. It leverages the [Dedot](https://github.com/dedotdev/dedot) library to interact with the networks and provide real-time editing, encoding and decoding of extrinsics.
+
+Our team is passionate about improving the developer experience in the Polkadot ecosystem. We believe that by simplifying the extrinsics building process, we can lower the barrier to entry for new developers and increase the efficiency of experienced ones.
+
+### Project Details
+
+<img src="https://raw.githubusercontent.com/itsyogesh/polkadot-hackathon-2024/refs/heads/main/singapore/27-Relaycode/docs/demo.gif" alt="Relaycode Demo" width="500">
```
Comment (PieWol) [inline applications/relaycode.md:27]: please don't use any html tags.

File: applications/relaycode.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+- **Estimated Duration:** 1 month
+- **FTE:** 2
+- **Costs:** 15,000 USD
+- **DOT %:** 50%
+
+|  Number | Deliverable                   | Specification                                                                                                                                            |
+| ------: | ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **0a.** | License                       | Apache 2.0                                                                                                                                               |
+| **0b.** | Documentation                 | We will provide basic documentation including: <br>- Inline code documentation<br>- A simple tutorial explaining how to use the MVP version of Relaycode |
```
Comment (PieWol) [inline applications/relaycode.md:136]: please don't use any html tags.

File: applications/relaycode.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+- **Estimated Duration:** 1 month
+- **FTE:** 2
+- **Costs:** 15,000 USD
+- **DOT %:** 50%
+
+|  Number | Deliverable                   | Specification                                                                                                                                                                                                 |
+| ------: | ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **0a.** | License                       | Apache 2.0                                                                                                                                                                                                    |
+| **0b.** | Documentation                 | We will provide comprehensive documentation including: <br>- Detailed inline code documentation<br>- An in-depth tutorial explaining all features of Relaycode<br>- API specifications for core functionality |
```
Comment (PieWol) [inline applications/relaycode.md:156]: same here

**Comment by semuelle** [comment]
Just to clarify @PieWol's comments: it's possible to use HTML tags, but they need to be properly closed, so `<br>` should be `<br />`, etc.

File: applications/relaycode.md (hunk 2 above)
Comment (itsyogesh) [reply applications/relaycode.md]: Added changes in https://github.com/w3f/Grants-Program/pull/2410/commits/b4118a1b4dcae827976ee280fbf0339e9ccf7504

**Comment by itsyogesh** [comment]
Hi @PieWol @semuelle thanks for pointing it out, I've added compatible markdown so its clear of any html tags!

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Thanks @itsyogesh would you be willing to switch to USDC instead of USDT on the Asset Hub? We just made a change to our payment options.

**Comment by itsyogesh** [comment]
> Thanks @itsyogesh would you be willing to switch to USDC instead of USDT on the Asset Hub? We just made a change to our payment options.

Hey @keeganquigley, no worries. Just updated the payment method to USDC.

**Review by keeganquigley — APPROVED** [review]
Thanks @itsyogesh I'm willing to give it a shot. While waiting for other reviewers, we require [KYC/KYB](https://in.sumsub.com/idensic/l/#/uni_p4ZARQeouBz5eHSt) for all grantees. Could you please submit your legal entity for verification along with your personal profile on the above link?

Thanks! Feel free to ping me on Element if you run into any issues.

**Review by Noc2 — APPROVED** [review]
Happy to go ahead with it as well. It would be nice to also see some documentation for the different extrinsics or at least to have a link to the documentation.

**Comment by itsyogesh** [comment]
> Happy to go ahead with it as well. It would be nice to also see some documentation for the different extrinsics or at least to have a link to the documentation.

Hey @Noc2 thank you for your support. Figuring out a way to abstract some of it out from the rust pallet docs, dedot does it automatically at pallet and extrinsic level but lacks documentation for arguments. In the meantime, I got the project running in its current state. You can check it out here https://relaycode.org.

**Review by laboon — APPROVED** [review]

**Comment by semuelle** [comment]
Hi @itsyogesh, since you are applying for this grant as a company, I have to ask you to submit the KYB form as well. The link provided above was only for KYC and is not actually applicable in this case. Sorry for the confusion.

**Comment by keeganquigley** [comment]
Hi @itsyogesh since it's been a year since the last milestone has been delivered, are you still working on this grant? Unless you are planning to submit a delivery in the next few weeks we will cancel it do to inactivity. Thank you.

**Pull request merged**